src/Servers.jl

```diff
@@ -394,9 +399,6 @@ function handle_connection(f::Function, c::Connection;
             io = Transaction(c)
             handle_transaction(f, io; final_transaction=(count == reuse_limit),
                                       kw...)
-            if count == reuse_limit
```

Comment:
If `reuse_limit=0`, then the connection was being closed before the handler could respond. Moved the close further down below.